tests/e2e/tidbcluster/tidbcluster.go

```diff
@@ -448,7 +459,17 @@ var _ = ginkgo.Describe("[tidb-operator] TiDBCluster", func() {
 			}),
 		}
 
-		oldPumpSet, err := c.AppsV1().StatefulSets(tc.Namespace).Get(controller.PumpMemberName(tc.Name), metav1.GetOptions{})
+		// If using advanced statefulset, we must upgrade all Kubernetes statefulsets to advanced statefulsets first.
+		if ocfg.Enabled(features.AdvancedStatefulSet) {
+			stsList, err := c.AppsV1().StatefulSets(ns).List(metav1.ListOptions{})
+			framework.ExpectNoError(err)
+			for _, sts := range stsList.Items {
+				_, err = helper.Upgrade(c, asCli, &sts)
+				framework.ExpectNoError(err)
+			}
+		}
```

Comment:
in this e2e case, we must upgrade sts deployed by helm if the advanced statefulset is ena bled